File: src/main/Java/org/team1085/robot/commands/DriveFor.java
package org.team1085.robot.commands;

import org.team1085.robot.Robot;

import edu.wpi.first.wpilibj.command.PIDCommand;

public class DriveFor extends PIDCommand {

    public DriveFor(double dist) {
        super(4.0, 20.0, 5.0);
        requires(Robot.chassis);
        setSetpoint(dist);
    }

    public void initialize() {
        Robot.chassis.resetEncoders();
    }

    protected double returnPIDInput() {
        return Robot.chassis.getRightEncoder();
    }

    protected void usePIDOutput(double value) {
        Robot.chassis.setSpeed(value);
    }

    protected boolean isFinished() {
        return false;
    }
}

File: src/main/kotlin/org/team6879/robot/commands/DriveFor.kt
package org.team6879.robot.commands

import edu.wpi.first.wpilibj.command.PIDCommand
import org.team6879.robot.Robot

class DriveFor (dist: Double) : PIDCommand (4.0, 20.0, 5.0) {

  init {
    requires(Robot.chassis)
    setSetpoint(dist)
  }

  override fun returnPIDInput () = Robot.chassis.getRightEncoder()
  override fun usePIDOutput (value: Double) = Robot.chassis.setSpeed(value)

  override fun initialize () = Robot.chassis.resetEncoders()
  override fun isFinished () = false

}